python/mp/effects/effect.py

```diff
@@ -20,26 +20,28 @@ class Effect(abc.ABC):
     # Can't decorate with @self.r, so need this here
     dm = DispatcherMapper()
 
-    def __init__(self, name, set, color=color.Color(1,1,1), duration=None):
+    def __init__(self, *args, **kwargs):
+
+        # Introducting...the option to pass a rosary on init instead of
+        # assigning it afterwards
+        self.rosary = kwargs.get('rosary')
         # the name is used when the Effect is registered
-        self.name = name
+        self.name = kwargs.get('name')
         # the bead_set is a list of beads the Effect is applied to. The order is important!
-        self.bead_set = self.set_bead_set(set)
+        self.bead_set = self.set_bead_set(kwargs.get('bead_set', set()))
         # The color of the Effect. This is not always meaningful.
-        self.color = copy.copy(color)
-        self.duration = duration
+        self.color = kwargs.get('color')
+        self.duration = kwargs.get('duration')
+        self.delay = kwargs.get('delay', 0)
```

Comment:
No longer specifying initialization parameters, and just trying them in **kwargs. This lets me pass generic parameters like `duration`, `delay`, and `rosary` in any effect's `__init__` method without explicitly declaring them in the subclasses.

The price we pay is that I'm now requiring all effect classes to take and pass along a **kwargs.

Actually...we could technically probably get away with never passing `name`, `bead_set`, or `color` in the individual effects too, and just let them fall through in **kwargs.

The downside to that is that we're left with a very minimal `__init__` and you "just have to know" what to pass.